plugins/modules/idrac_server_config_profile.py

```diff
@@ -669,11 +657,23 @@ def run_export_import_scp_http(idrac, module):
                                         job_wait=module.params["job_wait"], share=share,
                                         include_in_export=include_in_export)
     scp_response = response_format_change(scp_response, module.params, scp_file_name_format)
-    if isinstance(scp_response, dict) and scp_response.get("TaskStatus") == "Critical":
-        module.fail_json(msg="Failed to {0} scp.".format(command), scp_status=scp_response)
+    exit_on_failure(module, scp_response, command)
     return scp_response
 
 
+def perform_check_mode(module, idrac, job_wait, http_share=True):
+    if module.check_mode:
+        module.params["job_wait"] = True
```

Comment:
Why are we changing values of job_wait inside perform check mode, this function should be only for checking mode.

I can see on line 667, we are calling preview_scp_redfish, which is inside this check mode, I think it's better to call this function before calling perform_check_mode and pass the response to perform_check_mode.

Reply:
these are part of fixing CC. logic is same as existing check mode workflow.

Reply:
I got it Jennifer, May I know what's significance of line 665 and 666, where we are changing job_wait as true if it's check_mode and in last again reassigning it, any reason ?

Reply:
yes, needed to avoid job wait. Remove the unreachable reassignment line from method